### PR TITLE
Change pending cluster state to handle multiple bucket spaces

### PR DESCRIPTION
**Pull request by toregge**
@geirst, @vekterli: please review

### CODE REVIEW

File: storage/src/vespa/storage/distributor/pendingclusterstate.cpp [hunk 1]
```diff
@@ -64,19 +64,26 @@ PendingClusterState::PendingClusterState(
...
 {
     logConstructionInformation();
     markAllAvailableNodesAsRequiringRequest();
+    constructorHelper();
+}
+
+PendingClusterState::~PendingClusterState() {}
+
+void
+PendingClusterState::constructorHelper()
```
Comment (geirst) [inline storage/src/vespa/storage/distributor/pendingclusterstate.cpp]: Consider rename to something like "considerRequestBucketInfo"

Comment (geirst) [reply storage/src/vespa/storage/distributor/pendingclusterstate.cpp]: Or "init pending transitions"

File: storage/src/vespa/storage/distributor/pendingclusterstate.cpp [hunk 2]
```diff
@@ -64,19 +65,28 @@ PendingClusterState::PendingClusterState(
...
       _sender(sender),
+      _bucketSpaceRepo(bucketSpaceRepo),
       _bucketOwnershipTransfer(true),
-      _pendingTransition()
+      _pendingTransitions()
 {
     logConstructionInformation();
     markAllAvailableNodesAsRequiringRequest();
-    _pendingTransition = std::make_unique<PendingBucketSpaceDbTransition>(*this, _clusterInfo, _newClusterState, _creationTimestamp);
+    constructorHelper();
```
Comment (vekterli) [inline storage/src/vespa/storage/distributor/pendingclusterstate.cpp]: I'd prefer a more descriptive name here. Maybe `initializeBucketSpaceTransitions` or `sendInitialBucketSpaceRequests` ?

**Review by vekterli — APPROVED** [review]
:+1: see naming and logging nitpicks

**Review by geirst — APPROVED** [review]
LGTM. Please consider my comments

**Review by geirst — APPROVED** [review]
👍

**Pull request merged**